src/Common/IconCache.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Smuxi - Smart MUltipleXed Irc
+//
+// Copyright (c) 2015 Carlos Martín Nieto
+//
+// Full GPL License: <http://www.gnu.org/licenses/gpl.txt>
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
+using System;
+using System.IO;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Collections.Generic;
+using System.Linq;
+using Smuxi.Common;
+
+namespace Smuxi.Common
+{
+    public class IconCache
+    {
+        #if LOG4NET
+        private static readonly log4net.ILog f_Logger = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+        #endif
+
+        public IWebProxy Proxy { get; set; }
+
+        readonly string cachePath = Platform.CachePath;
+        readonly string iconsPath;
+
+        /// <summary>
+        /// Try to get a cached icon
+        /// </summary>
+        /// <returns><c>true</c>, if get icon is cached, <c>false</c> otherwise.</returns>
+        /// <param name="protocol">The protocol of the channel or server</param>
+        /// <param name="iconName">Name of the icon, including file extension</param>
+        /// <param name="value">Path to the cached icon, if cached</param>
+        public bool TryGetIcon(string protocol, string iconName, out string value)
```

Comment:
you added some XML markup for API docs in a nice way. please also mark thread-safe methods (this is good practice)